src/components/Admin/Authentication/Util.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import {
+  Button,
+  FormControl,
+  FormErrorMessage,
+  CircularProgress,
+} from '@chakra-ui/react';
+
+/**
+ * Props for SubmitButton component. Used for type safety.
+ */
+interface SubmitButtonProps {
+  label?: string;
+  isLoading?: boolean;
+  color?: string;
+  error?: string;
+  isDisabled?: boolean;
+}
+
+/**
+ * Component for submit button in authentication forms.
+ * @param props - label, isLoading, color, error, isDisabled
+ * - `label?: string` label for button. Defaults to "Submit"
+ * - `isLoading?: boolean` if application is currently loading, used to display
+ *   loading circle on button. Defaults to false
+ * - `color?: string` color of button. Defaults to "teal"
+ * - `error?: string` error message to be displayed. Defaults to ''
+ * - `isDisabled?: boolean` if button is clickable. Defaults to false
```

Comment:
Nit: we don't usually include the types in our doc strings (unless you think it's important that they are included here)

Reply:
I agree, since TypeScript already has types, adding types in the documentation is repetitive.